tests/optim/test_oss.py

```diff
@@ -843,13 +835,21 @@ def closure_sharded(input_tensor=input_tensor):
                 loss_ddp, loss_sharded_optim
             ), f"Losses differ in between Pytorch optim and OSS\nworld size {world_size}"
 
+            check_same_model_params(oss_ddp_model, ddp_model)
+
         # The model should be synchronized in between the ranks at construction time, check that
-        check_same_model_params()
+        check_same_model_params(oss_ddp_model, ddp_model)
 
         # The models should stay the same in between ddp and sharded optimizer
         for i in range(5):
             check_step()
-            check_same_model_params()
+
+            # Check that altering the trainable parameters does not cause DDP and OSS to diverge
+            if change_train_graph:
```

Comment:
new logic change in this test, check that changing trainability is properly taken into account. Commenting out this PRs additions in OSS/ShardedDDP breaks, as expected